src/sentry/sentry_apps/api/endpoints/sentry_app_installations.py

```diff
@@ -90,6 +92,9 @@ def post(self, request: Request, organization) -> Response:
                 sentry_app__slug=slug, organization_id=organization.id
             )
         except SentryAppInstallation.DoesNotExist:
+            assert isinstance(request.user, User) or isinstance(
+                request.user, RpcUser
+            ), "user must be authenticated to create a SentryAppInstallation"
```

Comment:
is this typing too narrow? -> are there other types of users (RpcUser, User, AnonymousUser)? Is it even true that being authenticated == being a User/RpcUser.

Reply:
> are there other types of users (RpcUser, User, AnonymousUser)? 

No, those are all the type options for `request.user`

> Is it even true that being authenticated == being a User/RpcUser.

Yes, if a request is not authenticated `request.user` will be `AnonymousUser`.